shared/src/error/mod.rs

```diff
@@ -1,101 +1,116 @@
+    /// (including Ok).
+    pub fn is_system_error(self) -> bool {
+        self.value < (Self::FIRST_UNRESERVED_EXIT_CODE)
+    }
+}
 
-    /// Indicates a message execution is forbidden for the caller.
-    SysErrForbidden = 8,
+impl std::fmt::Display for ExitCode {
+    fn fmt(&self, f: &mut Formatter<'_>) -> std::fmt::Result {
+        write!(f, "{}", self.value)
+    }
+}
 
-    /// Indicates actor code performed a disallowed operation. Disallowed operations include:
-    /// - mutating state outside of a state acquisition block
-    /// - failing to invoke caller validation
-    /// - aborting with a reserved exit code (including success or a system error).
-    SysErrIllegalActor = 9,
+/// Enumerates exit codes which originate inside the VM.
+/// These values may not be used by actors when aborting.
+pub struct SystemExitCode {}
 
-    /// Indicates an invalid argument passed to a runtime method.
-    SysErrIllegalArgument = 10,
+impl SystemExitCode {
+    /// Indicates the message sender doesn't exist.
+    pub const SENDER_INVALID: ExitCode = ExitCode::new(1);
+    /// Indicates that the message sender was not in a valid state to send this message.
+    /// Either:
+    /// - The sender's nonce nonce didn't match the message nonce.
+    /// - The sender didn't have the funds to cover the message gas.
+    pub const SENDER_STATE_INVALID: ExitCode = ExitCode::new(2);
+    /// Indicates failure to find a method in an actor.
+    pub const INVALID_METHOD: ExitCode = ExitCode::new(3); // FIXME: reserved
+    /// Indicates the message receiver trapped (panicked).
+    pub const ILLEGAL_INSTRUCTION: ExitCode = ExitCode::new(4);
+    /// Indicates the message receiver doesn't exist and can't be automatically created
+    pub const INVALID_RECEIVER: ExitCode = ExitCode::new(5);
+    /// Indicates the message sender didn't have the requisite funds.
+    pub const INSUFFICIENT_FUNDS: ExitCode = ExitCode::new(6);
+    /// Indicates message execution (including subcalls) used more gas than the specified limit.
+    pub const OUT_OF_GAS: ExitCode = ExitCode::new(7);
+    // REVIEW: I restored this in order to map the syscall error number ErrorNumber::IllegalOperation
+    // Should it use ILLEGAL_INSTRUCTION instead?
+    pub const ILLEGAL_ACTOR: ExitCode = ExitCode::new(8);
+    /// Indicates the message receiver aborted with a reserved exit code.
+    pub const ILLEGAL_EXIT_CODE: ExitCode = ExitCode::new(9);
+    /// Indicates an internal VM assertion failed.
+    pub const ASSERTION_FAILED: ExitCode = ExitCode::new(10);
+    /// Indicates the actor returned a block handle that doesn't exist
+    pub const MISSING_RETURN: ExitCode = ExitCode::new(11);
+
+    pub const RESERVED_12: ExitCode = ExitCode::new(12);
+    pub const RESERVED_13: ExitCode = ExitCode::new(13);
+    pub const RESERVED_14: ExitCode = ExitCode::new(14);
+    pub const RESERVED_15: ExitCode = ExitCode::new(15);
```

Comment:
I'd document but remove these. Otherwise, using them would be a breaking change.